Source/Core/VideoCommon/ShaderCache.cpp

```diff
@@ -15,14 +15,19 @@
 #include "VideoCommon/Statistics.h"
 #include "VideoCommon/VertexLoaderManager.h"
 #include "VideoCommon/VertexManagerBase.h"
+#include "VideoCommon/VideoCommon.h"
+#include "VideoCommon/VideoConfig.h"
 
-#include "imgui.h"
+#include <imgui.h>
 
 std::unique_ptr<VideoCommon::ShaderCache> g_shader_cache;
 
 namespace VideoCommon
 {
-ShaderCache::ShaderCache() = default;
+ShaderCache::ShaderCache() : m_api_type{APIType::Nothing}
```

Comment:
Feels a bit weird to go from direct member initialization and `default` ctor to one that's typed out just to have a forward-declared `enum class` in the header instead. Is this really worth it?

Reply:
I don't particularly see anything unusual about it in this case, given if anyone was concerned about the final state of the internal member variables after construction, then one would need to see what the constructor is doing anyways. I can revert this portion of the change if it's not desirable.

Reply:
No, was mainly wondering since for the most part, the suggestion is to do the reverse (get rid of the ctor and move it to the members themselves; so it isn't forgotten if another ctor is added).
Just wanting to wrap my head around it.

Reply:
I think in this case it's more readable to keep it in the header - it doesn't seem to remove any includes?

Reply:
It doesn't remove any includes in the header because it was relying on an indirect include previously